executor/include/executor/trait/is_executor.hpp

```diff
@@ -21,19 +21,30 @@ struct is_executor : std::false_type {};
 
 namespace detail {
 const auto noop = [] {};
-}
 
+template <typename Executor, typename = void>
+struct contains_execute : std::false_type {};
+
+template <typename Executor>
+struct contains_execute<
+    Executor, std::enable_if_t<std::is_same<
+                  decltype(std::declval<executor::remove_cv_ref_t<Executor>>()
+                               .execute(detail::noop)),
+                  void>::value>> : std::true_type {};
+
+template <typename Executor>
+using check_equality_comparable =
+    decltype(std::declval<Executor>() == std::declval<Executor>());
```

Comment:
I think this will either fail to compile, or simply `check_equality_comparable` to bool. At some point in your code, `check_equality_comparable` needs to evaluate to true or false. It seems you're interested in something which is true if `std::declval<T>() == std::declval<T>()` exist and false, in case it is not defined. 

See https://godbolt.org/z/Qvvx9e

```cpp
template <typename T1, typename T2, typename = void>
struct is_equality_comparable : std::false_type{};

template <typename T1, typename T2>
struct is_equality_comparable<
    T1,
    T2,
    std::void_t<decltype(std::declval<T1>() == std::declval<T2>())>
> : std::true_type{};
```

Reply:
Official named requirement: https://en.cppreference.com/w/cpp/concepts/equality_comparable

Reply:
4d741f3 should resolve this